src/sync_protocol/include/agent_sync_protocol.hpp

```diff
@@ -134,7 +132,61 @@ class AgentSyncProtocol : public IAgentSyncProtocol
         /// @param fbData Flatbuffer data
         /// @param module Module name
         /// @return True on success, false on failure
-        bool sendFlatBufferMessageAsString(flatbuffers::span<uint8_t> fbData, const std::string& module);
+        bool sendFlatBufferMessageAsString(const std::vector<uint8_t>& fbData, const std::string& module);
+
+        /// @brief Defines the possible phases of a synchronization process.
+        enum class SyncPhase
+        {
+            /// @brief The protocol is not in an active synchronization process.
+            Idle,
+            /// @brief A start message has been sent, waiting for the manager's StartAck.
+            WaitingStartAck,
+            /// @brief An end message has been sent, waiting for the manager's EndAck.
+            WaitingEndAck
+        };
+
+        /// @brief Synchronization state shared between threads during module sync.
+        ///
+        /// This structure holds synchronization primitives and state flags used to
+        /// coordinate between the main synchronization thread and the response handler.
+        /// It stores whether specific acknowledgments have been received and the ranges
+        /// requested by the manager.
+        struct SyncState
+        {
+            /// @brief Mutex used to protect access to the synchronization state.
+            std::mutex mtx;
+
+            /// @brief Condition variable used to signal waiting threads.
+            std::condition_variable cv;
+
+            /// @brief Indicates whether a StartAck response has been received.
+            bool startAckReceived = false;
+
+            /// @brief Indicates whether an EndAck response has been received.
+            bool endAckReceived = false;
+
+            /// @brief Ranges requested by the manager via ReqRet message.
+            std::vector<std::pair<uint64_t, uint64_t>> reqRetRanges;
+
+            /// @brief Current phase of the synchronization process.
+            SyncPhase phase = SyncPhase::Idle;
+
+            /// @brief Unique identifier for the current synchronization session, received from the manager.
+            uint64_t session = 0;
+
+            /// @brief Resets all internal flags and clears received ranges.
+            ///
+            /// This should be called before starting a new synchronization cycle.
+            void reset()
```

Comment:
What about `sync` and `session`? Don't they need to be reset?

Reply:
Added!